tests/api.c

```diff
@@ -61823,6 +61826,102 @@ static int test_wolfSSL_CRL_CERT_REVOKED_alert(void)
 }
 #endif
 
+#if defined(WOLFSSL_DTLS) && !defined(WOLFSSL_NO_TLS12) &&          \
+    !defined(NO_WOLFSSL_CLIENT) && !defined(NO_WOLFSSL_SERVER) &&   \
+    !defined(SINGLE_THREADED)
+
+static void test_dtls_msg_from_other_peer_set_get_port(word16 *p, int get)
+{
+    static word16 port;
+
+    if (get)
+        *p = port;
+    else
+        port = *p;
+}
+
+static int test_dtls_msg_from_other_peer_cb(WOLFSSL_CTX *ctx, WOLFSSL *ssl)
+{
+    char buf[1] = {'t'};
+    SOCKADDR_IN_T addr;
+    int sock_fd;
+    word16 port;
+    int err;
+
+    (void)ssl;
+    (void)ctx;
+
+    test_dtls_msg_from_other_peer_set_get_port(&port, 1);
+
+    sock_fd = socket(AF_INET_V, SOCK_DGRAM, 0);
+    if (sock_fd == -1)
+        return -1;
+    build_addr(&addr, wolfSSLIP, port, 1, 0);
+    /* send a packet to the server. Being another socket, the kernel will ensure
+     * the source port will be different. */
+
+    err = (int)sendto(sock_fd, buf, sizeof(buf), 0, (SOCKADDR*)&addr,
+        sizeof(addr));
+    if (err == -1)
+        return -1;
+
+    close(sock_fd);
+    return 0;
+}
+
+static int test_dtls_msg_from_other_peer(void)
+{
+    callback_functions client_cbs;
+    callback_functions server_cbs;
+    THREAD_TYPE serverThread;
+    func_args client_args;
+    func_args server_args;
+    tcp_ready ready;
+
+    XMEMSET((byte*)&client_args, 0, sizeof(func_args));
+    XMEMSET((byte*)&server_args, 0, sizeof(func_args));
+    XMEMSET((byte*)&client_cbs, 0, sizeof(client_cbs));
+    XMEMSET((byte*)&server_cbs, 0, sizeof(server_cbs));
+
+    StartTCP();
+    InitTcpReady(&ready);
+
+#if defined(USE_WINDOWS_API)
+    /* use RNG to get random port if using windows */
+    ready.port = GetRandomPort();
+#endif
+
+    client_cbs.method = wolfDTLSv1_2_client_method;
+    server_cbs.method = wolfDTLSv1_2_server_method;
+    client_cbs.doUdp = 1;
+    server_cbs.doUdp = 1;
+    server_args.signal = &ready;
+    server_args.callbacks = &server_cbs;
+    client_args.signal = &ready;
+    client_args.callbacks = &client_cbs;
+
+    start_thread(test_server_nofail, &server_args, &serverThread);
+    wait_tcp_ready(&server_args);
+    test_dtls_msg_from_other_peer_set_get_port(&ready.port, 0);
+    test_client_nofail(&client_args,
+        test_dtls_msg_from_other_peer_cb);
+    join_thread(serverThread);
```

Comment:
If possible please use `test_wolfSSL_client_server_nofail`. It greatly simplifies the tests.